README.md

```diff
@@ -57,38 +57,74 @@ helm delete --purge che-plugin-registry
-| firstPublicationDate | date of publishing the plugin (in ISO 8601) | no<sup>2</sup> |
-| latestUpdateDate | date of latest plugin update (in ISO 8601) | no<sup>3</sup> |
-| preview | a URL to devfile, to preview this plugin | no |
-| tags | a list of tags, related to this plugin | no|
-| mediaImage | links for images showcasing the plugin | no |
-| mediaVideo | links for video showcasing the plugin| no |
-| attributes | a map of special attributes, can be used for instance, imported plugins from VS Code | no |
+Here is an overview of all fields that can be present in plugin meta YAML files. This document represents the current `v3` version.
+
+```yaml
+apiVersion:            # v2; v1 supported for backwards compatability
+publisher:             # publisher name
+name:                  # plugin name
+version:               # plugin version
+type:                  # plugin type; e.g. "Theia plugin", "Che Editor"
+displayName:           # name shown in user dashboard
+title:                 # plugin title
+description:           # short description of plugin's purpose
+icon:                  # link to SVG icon
+repository:            # repo hosting plugin sources
+category:              # see [1]
+firstPublicationDate:  # optional; see [2]
+latestUpdateDate:      # optional; see [3]
+deprecate:             # optional; section for deprecating plugins in favor of others
+  autoMigrate:         # boolean
+  migrateTo:           # new plugin id
+spec:                  # spec (used to be che-plugin.yaml)
+  endpoints:           # optional; plugin endpoints
+    - name:
+      public:            # boolean
+      targetPort:
+      attributes:
+        protocol:        # protocol used for communicating over endpoint
+        secure:          # use secure version of protocol above
+        discoverable:    # if false, no service is created for this endpoint
+        cookiesAuthEnabled:
+        type:
+        path:
+  containers:          # optional; sidecar containers for plugin
+    - image:
+      name:              # name used for sidecar container
+      memorylimit:       # Kubernetes/OpenShift-spec memory limit string (e.g. "512Mi")
+      env:               # list of env vars to set in sidecar
+        - name:
+          value:
+      volumes:           # volumes required by plugin
+          - mountPath:
+            name:
+      ports:             # ports exposed by plugin
+        - exposedPort:
+      commands:          # commands available to plugin container
+        - name:
+          workingDir:
+          command:       # list of commands + arguments, e.g.:
+            - rm
+            - -rf
+            - /cache/.m2/repository
+      mountSources:      # boolean
+  workspaceEnv:        # optional; env vars for the workspace
+    - name:
+      value:
+  extensions:            # optional; required for VS Code/Theia plugins; list of urls to plugin artifacts (.vsix/.theia files) -- examples follow
```

Comment:
This is input to the plugin broker, right? How can the user find out which plugin file formats are supported? Shouldn't we mention the restriction on downloading from the vs code marketplace here?

Reply:
There aren't currently many restrictions here; the broker will download whatever you give it. 

I hesitate to codify the VS Code marketplace limitation here but can add it if it is important.